src/util/misc/isTransparent.ts

```diff
@@ -13,39 +15,26 @@ export const isTransparent = (
   y: number,
   tolerance: number
 ): boolean => {
-  // If tolerance is > 0 adjust start coords to take into account.
-  // If moves off Canvas fix to 0
-  if (tolerance > 0) {
-    if (x > tolerance) {
-      x -= tolerance;
-    } else {
-      x = 0;
-    }
-    if (y > tolerance) {
-      y -= tolerance;
-    } else {
-      y = 0;
-    }
+  tolerance = Math.ceil(Math.max(tolerance, 0));
+  const point = new Point(x, y);
+  const start = point.scalarSubtract(tolerance).floor();
+  const end = point
+    .scalarAdd(Math.max(tolerance, 1))
+    .ceil()
+    .min(new Point(ctx.canvas.width - 1, ctx.canvas.height - 1));
+  const boundStart = start.max(new Point());
+  const size = end.subtract(boundStart);
+  if (size.x <= 0 || size.y <= 0) {
+    // out of bounds
+    return true;
```

Comment:
i notice now this change, i don't think we should do it. the old loop is so much clear and smaller.
There is something weird in the old code and is that if we set x or y to 0 we do not adjust the tolerance to avoid considering extra pixels, but this change here is definitely extra complication.

Reply:
I will try to make simple but seems to me there are edge cases in the logic.
I can't understand the logic

Reply:
please try to explain it